src/overlays/actors/ovl_Demo_Ext/z_demo_ext.c

```diff
@@ -1,17 +1,233 @@
+/*
+ * File: z_demo_ext.c
+ * Overlay: Demo_Ext
+ * Description: Magic Vortex in Silver Gauntlets Cutscene
+ */
+
 #include "z_demo_ext.h"
+#include "vt.h"
 
 #define FLAGS 0x00000010
 
 #define THIS ((DemoExt*)thisx)
 
+typedef enum {
+    /* 0x00 */ EXT_WAIT,
+    /* 0x01 */ EXT_MAINTAIN,
+    /* 0x02 */ EXT_DISPELL
+} DemoExtAction;
+
+typedef enum {
+    /* 0x00 */ EXT_DRAW_NOTHING,
+    /* 0x01 */ EXT_DRAW_VORTEX
+} DemoExtDrawMode;
+
 void DemoExt_Init(Actor* thisx, GlobalContext* globalCtx);
 void DemoExt_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void DemoExt_Update(Actor* thisx, GlobalContext* globalCtx);
 void DemoExt_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-extern UNK_TYPE D_0600FAA0;
+extern Gfx D_0600FAA0[];
+
+void DemoExt_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void DemoExt_Init(Actor* thisx, GlobalContext* globalCtx) {
+    DemoExt* this = THIS;
+
+    this->scrollIncr[0] = 25;
+    this->scrollIncr[1] = 40;
+    this->scrollIncr[2] = 5;
+    this->scrollIncr[3] = 30;
+    this->primAlpha = kREG(28) + 0xFF;
+    this->envAlpha = kREG(32) + 0xFF;
+    this->scales.x = kREG(19) + 400.0f;
+    this->scales.y = kREG(20) + 100.0f;
+    this->scales.z = kREG(21) + 400.0f;
+}
+
+void DemoExt_PlayVortexSFX(DemoExt* this) {
+    if (this->alphaTimer <= (kREG(35) + 40.0f) - 15.0f) {
+        Audio_PlaySoundGeneral(NA_SE_EV_FANTOM_WARP_L - SFX_FLAG, &this->actor.projectedPos, 4, &D_801333E0,
+                               &D_801333E0, &D_801333E8);
+    }
+}
+
+CsCmdActorAction* DemoExt_GetNpcCsAction(GlobalContext* globalCtx, s32 npcActionIndex) {
+    if (globalCtx->csCtx.state != 0) {
```

Comment:
can name it GetNpcAction if you like, with a name like that its implied its for a cutscene
and I believe theres an enum for cutscene state, but I dont think we use it everywhere yet. if not then ignore

Reply:
Well, this actor does only appear in a cutscene. Is that not quite what you had in mind?

Reply:
NpcAction is already a structure specifically for cutscenes, so I was noting that having `Cs` in the name isnt really neeeded

Reply:
Ah, okay